[19.07] https-dns-proxy: recompile with -fPIC

Maintainer: @stangri 
Environment: aarch64_cortex-a53 (Turris MOX), OpenWrt 19.07.8

Config:
```
CONFIG_PKG_FORTIFY_SOURCE_2=y
CONFIG_PKG_ASLR_PIE_ALL=y
CONFIG_PKG_CC_STACKPROTECTOR_STRONG=y
CONFIG_PKG_RELRO_FULL=y
```

Description:

```
[ 37%] Building C object CMakeFiles/https_dns_proxy.dir/src/https_client.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c: In function 'write_buffer':
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:53:30: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
     ELOG_REQ("Out of memory!");
                              ^
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c: In function 'https_fetch_ctx_process_response':
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:297:49: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
                  "but no response was sent back");
                                                 ^
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:307:82: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
         WLOG_REQ("No response (probably connection has been closed or timed out)");
                                                                                  ^
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:383:48: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
           DLOG_REQ("CURLINFO_HTTP_VERSION: 1.0");
                                                ^
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:386:48: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
           DLOG_REQ("CURLINFO_HTTP_VERSION: 1.1");
                                                ^
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:389:46: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
           DLOG_REQ("CURLINFO_HTTP_VERSION: 2");
                                              ^
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:420:38: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
       ELOG_REQ("Error getting timing");
                                      ^
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c: In function 'https_fetch_ctx_cleanup':
/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29/src/https_client.c:442:60: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
         ILOG_REQ("Response was faulty, skipping DNS reply.");
                                                            ^
[ 50%] Building C object CMakeFiles/https_dns_proxy.dir/src/logging.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 62%] Building C object CMakeFiles/https_dns_proxy.dir/src/main.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 75%] Building C object CMakeFiles/https_dns_proxy.dir/src/options.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 87%] Building C object CMakeFiles/https_dns_proxy.dir/src/stat.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[100%] Linking C executable https_dns_proxy
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-7.5.0_musl/lib/gcc/aarch64-openwrt-linux-musl/7.5.0/../../../../aarch64-openwrt-linux-musl/bin/ld: CMakeFiles/https_dns_proxy.dir/src/options.c.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `optarg' which may bind externally can not be used when making a shared object; recompile with -fPIC
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-7.5.0_musl/lib/gcc/aarch64-openwrt-linux-musl/7.5.0/../../../../aarch64-openwrt-linux-musl/bin/ld: CMakeFiles/https_dns_proxy.dir/src/options.c.o(.text+0x120): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `optarg'
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-7.5.0_musl/lib/gcc/aarch64-openwrt-linux-musl/7.5.0/../../../../aarch64-openwrt-linux-musl/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

This also affects OpenWrt 21.02 and OpenWrt daily snapshots.

21.02 log:
```
-- The C compiler identification is GNU 8.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/bin/aarch64-openwrt-linux-musl-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: /mox/build/staging_dir/host/bin/git (found version "2.20.1") 
-- Version: 2021.08.18-da3da22e1f
-- clang-tidy not found.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_ASM_COMPILER
    CMAKE_ASM_COMPILER_LAUNCHER
    CMAKE_CXX_COMPILER
    CMAKE_CXX_FLAGS_RELEASE
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_EXPORT_PACKAGE_REGISTRY
    CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY
    CMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_MODULE_LINKER_FLAGS
    CMAKE_SHARED_LINKER_FLAGS
    DL_LIBRARY


-- Build files have been written to: /mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29
make[3]: Entering directory '/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29'
make[4]: Entering directory '/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29'
make[5]: Entering directory '/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29'
Scanning dependencies of target https_dns_proxy
make[5]: Leaving directory '/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29'
make[5]: Entering directory '/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29'
[ 12%] Building C object CMakeFiles/https_dns_proxy.dir/src/dns_poller.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 25%] Building C object CMakeFiles/https_dns_proxy.dir/src/dns_server.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 37%] Building C object CMakeFiles/https_dns_proxy.dir/src/https_client.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 50%] Building C object CMakeFiles/https_dns_proxy.dir/src/logging.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 62%] Building C object CMakeFiles/https_dns_proxy.dir/src/main.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 75%] Building C object CMakeFiles/https_dns_proxy.dir/src/options.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[ 87%] Building C object CMakeFiles/https_dns_proxy.dir/src/stat.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[100%] Linking C executable https_dns_proxy
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: CMakeFiles/https_dns_proxy.dir/src/options.c.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `optarg' which may bind externally can not be used when making a shared object; recompile with -fPIC
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: CMakeFiles/https_dns_proxy.dir/src/options.c.o(.text+0x120): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `optarg'
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[5]: *** [CMakeFiles/https_dns_proxy.dir/build.make:193: https_dns_proxy] Error 1
```

daily snapshots (master branch):

```
-- The C compiler identification is GNU 8.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/bin/aarch64-openwrt-linux-musl-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Git: /mox/build/staging_dir/host/bin/git (found version "2.20.1") 
-- Version: 2021.08.04-c48ea7f1b1
-- clang-tidy not found.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_ASM_COMPILER
    CMAKE_ASM_COMPILER_LAUNCHER
    CMAKE_CXX_COMPILER
    CMAKE_CXX_FLAGS_RELEASE
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_EXPORT_PACKAGE_REGISTRY
    CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY
    CMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_MODULE_LINKER_FLAGS
    CMAKE_SHARED_LINKER_FLAGS
    DL_LIBRARY


-- Build files have been written to: /mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29
ninja: Entering directory `/mox/build/build_dir/target-aarch64_cortex-a53_musl/https-dns-proxy-2021-07-29'
[1/8] Building C object CMakeFiles/https_dns_proxy.dir/src/dns_poller.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[2/8] Building C object CMakeFiles/https_dns_proxy.dir/src/dns_server.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[3/8] Building C object CMakeFiles/https_dns_proxy.dir/src/https_client.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[4/8] Building C object CMakeFiles/https_dns_proxy.dir/src/logging.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[5/8] Building C object CMakeFiles/https_dns_proxy.dir/src/main.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[6/8] Building C object CMakeFiles/https_dns_proxy.dir/src/options.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[7/8] Building C object CMakeFiles/https_dns_proxy.dir/src/stat.c.o
cc1: note: someone does not honour COPTS correctly, passed 0 times
[8/8] Linking C executable https_dns_proxy
FAILED: https_dns_proxy 
: && /mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/bin/aarch64-openwrt-linux-musl-gcc -Wall -Wextra --pedantic -Wno-strict-aliasing -Wno-variadic-macros -g -DDEBUG -L/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/usr/lib -L/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/lib -DPIC -fPIC -specs=/mox/build/include/hardened-ld-pie.specs -znow -zrelro CMakeFiles/https_dns_proxy.dir/src/dns_poller.c.o CMakeFiles/https_dns_proxy.dir/src/dns_server.c.o CMakeFiles/https_dns_proxy.dir/src/https_client.c.o CMakeFiles/https_dns_proxy.dir/src/logging.c.o CMakeFiles/https_dns_proxy.dir/src/main.c.o CMakeFiles/https_dns_proxy.dir/src/options.c.o CMakeFiles/https_dns_proxy.dir/src/stat.c.o -o https_dns_proxy  -lcares  -lcurl  -lev  -lresolv && :
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: CMakeFiles/https_dns_proxy.dir/src/options.c.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `optarg' which may bind externally can not be used when making a shared object; recompile with -fPIC
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: CMakeFiles/https_dns_proxy.dir/src/options.c.o(.text+0x120): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `optarg'
/mox/build/staging_dir/toolchain-aarch64_cortex-a53_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/../../../../aarch64-openwrt-linux-musl/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

--- a/net/https-dns-proxy/Makefile
+++ b/net/https-dns-proxy/Makefile
@@ -2,7 +2,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=https-dns-proxy
 PKG_VERSION:=2021-07-29
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL:=https://github.com/aarond10/https_dns_proxy/

--- a/net/https-dns-proxy/patches/010-fix-cmakelists.patch
+++ b/net/https-dns-proxy/patches/010-fix-cmakelists.patch
@@ -1,0 +1,19 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -12,12 +12,12 @@ function(define_file_basename_for_source
+   endforeach()
+ endfunction()
+ 
+-set(CMAKE_BUILD_TYPE "Debug")
++#set(CMAKE_BUILD_TYPE "Debug")
+ #set(CMAKE_BUILD_TYPE "Release")
+ 
+-set(CMAKE_C_FLAGS "-Wall -Wextra --pedantic -Wno-strict-aliasing -Wno-variadic-macros")
+-set(CMAKE_C_FLAGS_DEBUG "-g -DDEBUG")
+-set(CMAKE_C_FLAGS_RELEASE "-O2")
++#set(CMAKE_C_FLAGS "-Wall -Wextra --pedantic -Wno-strict-aliasing -Wno-variadic-macros")
++#set(CMAKE_C_FLAGS_DEBUG "-g -DDEBUG")
++#set(CMAKE_C_FLAGS_RELEASE "-O2")
+ 
+ if ((CMAKE_C_COMPILER_ID MATCHES GNU   AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 9) OR
+     (CMAKE_C_COMPILER_ID MATCHES Clang AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 10))

